remote-config/src/fetch/single.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::fetch::{
+    ConfigApplyState, ConfigFetcher, ConfigFetcherState, ConfigInvariants, FileStorage, OpaqueState,
+};
+use crate::file_change_tracker::{Change, ChangeTracker, FilePath, UpdatedFiles};
+use crate::{RemoteConfigPath, Target};
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
+/// Simple implementation
+pub struct SingleFetcher<S: FileStorage> {
+    fetcher: ConfigFetcher<S>,
+    target: Arc<Target>,
+    runtime_id: String,
+    config_id: String,
+    opaque_state: OpaqueState,
+}
+
+impl<S: FileStorage> SingleFetcher<S> {
+    pub fn new(sink: S, target: Target, runtime_id: String, invariants: ConfigInvariants) -> Self {
+        SingleFetcher {
+            fetcher: ConfigFetcher::new(sink, Arc::new(ConfigFetcherState::new(invariants))),
+            target: Arc::new(target),
+            runtime_id,
+            config_id: uuid::Uuid::new_v4().to_string(),
```

Comment:
What does this config_id represent?

A given client doesn't know about its config ID's ahead of time, and that can constantly be changing depending on configurations being created or deleted by users.

Reply:
Renamed to client_id.